jsapp/js/api.ts

```diff
@@ -33,20 +34,24 @@ export function handleApiFail(response: FailResponse) {
     message = htmlDoc.getElementsByClassName('errormsg')?.[0]?.innerHTML;
   }
 
-  if (!message) {
-    message = t('An error occurred');
+  const htmlMessage = message;
```

Comment:
I get a little bit confused at first with the multiple different things that `message` can represent and how `htmlMessage` weaves in between them. Could we have some separately-named variables like, for example, a `responseMessage` that would be either the plain-text or extracted-from-HTML `errormsg` from the server? Then, `message` perhaps could be only what's being built up to display to the user.

Reply:
Good point! I ended up renaming `message` to `responseMessage`, since it gets initialized to `response.responseText`, and then moving things around a bit to fit. Let me know if that doesn't work.